### ISSUE TITLE
Empty template in string transpiles to JS with syntax errors. 

### ISSUE THREAD

**Issue by HeroicAutodidact** [Microsoft/TypeScript#12344]
**TypeScript Version**:  2.0.8 

**Code**

```ts
console.log(`${}`);
```

**Expected behavior:**
I would expect running `tsc <filename>` to either transpile an empty template into an empty string or to throw an error. 

**Actual behavior:**
```js
console.log('' + );
```

Which, of course, throws `SyntaxError: Unexpected token ) ` when executed. A confusing error when one is working with automated transpilation and source maps.

**Comment by rbuckton** [comment]
An empty template is a syntactic error. We do not guarantee valid syntactic output when the input is syntactically invalid.

**Comment by rbuckton** [comment]
Running `tsc` for the above example reports the error:
```
error TS1109: Expression expected.
```

**Comment by HeroicAutodidact** [comment]
I thought this might be the case, it's a sensible design decision and I respect it.

I do believe that the error should be shown by tsc _as a syntax error within typescript_ rather than letting it produce invalid JavaScript. Perhaps this was fixed in 2.1, the bug was discovered using 2.0.8.

Thanks for looking into it.
